src/core/pool.rs

```diff
@@ -64,7 +64,7 @@ impl<T> Pool<T> {
     pub fn acquire(&self) -> LeasedItem<T> {
         let generation = self.generation();
         loop {
-            let gen_item = self.queue.pop();
+            let gen_item = self.queue.pop().unwrap();
```

Comment:
Reading the documentation, the semantics are very different.

When the queue is empty, MsQueue was blocking on `.pop()`. `SegQueue`, on the other hand returns an error when it is empty.

We really want the earlier behavior here. The fact that it was not caught by unit test means... that we lack unit test :).
Possible way to test that. Create a pool with 2 elements. Have three threads pick an element and sleep for `1ms`. This should not panic.

Reply:
I have been thinking about how to test a blocking call. I have a commented out test below, trying to follow your logic, but I haven't found a good way of testing it